intel/init.py

```diff
@@ -172,6 +241,44 @@ def check_isolated_cores(platform, num_exclusive_cores, num_shared_cores):
                 (num_isolated_cores, required_isolated_cores))
 
 
+def check_sst_cp_isolated_cores(platform, num_exclusive_cores,
+                                num_shared_cores, epp_order):
+    error_occured = False
+
+    if len(epp_order) == 3:
```

Comment:
stupid question - why 3? I don't see `epp_order[2]` in that code block

I'm assuming it works correctly, just interested in what determines the length of that epp_order list

Reply:
SST-CP can be set up with either 3 EPP values or 2. When there's 3 EPP values, the cores with the highest EPP value will be put into the exclusive core, the cores with the second highest EPP value will be put into the shared, and the rest in infra. What happens when 2 EPP values are present is that the highest priority cores will be split among the exclusive and shared pools. So when checking isolated cores you have to check whether you need to take into account the second EPP value. And we don't need to check the infra cores so the final element of epp_order can be ignored. Hope that clears things up :)

Reply:
yup, that's clear, thanks!